### PR TITLE
Add UniCase support to phf_macros and bump unicase version

### PR DESCRIPTION
**Pull request by Bobo1239**
This is actually my first time doing anything compiler/plugin related so this may be pretty bad...
Any pointers to improvements are highly welcome.

I'm not sure if there's a better solution regarding the naming of the feature (related: https://github.com/rust-lang/cargo/issues/1286 by you :smile:)

### CODE REVIEW

**Comment by sfackler** [comment]
The approximate matching on the macro side is a bit unfortunate, but I can't think of a better way either. Seems reasonable to me!

**Comment by sfackler** [comment]
Can you add a compile-fail test to make sure equivalent keys (e.g. `UniCase("foo")` and `UniCase("FOO")`) are prohibited?

**Comment by Bobo1239** [comment]
Using [compiletest-rs](https://github.com/laumann/compiletest-rs)?

**Comment by sfackler** [comment]
Yep! I thought we already had some but I guess not. You can look here for example usage: https://github.com/sfackler/rust-postgres-derive/tree/master/postgres-derive-macros/tests

**Comment by Bobo1239** [comment]
Not sure if there's another way to run certain compiletests only when a feature is activated.

**Comment by sfackler** [comment]
I don't see any reason to flag the compiletests off - let's just always have them active.

**Comment by Bobo1239** [comment]
I did that so running plain `cargo test` wouldn't result in:
```
error: tests/compile-fail-unicase/equivalent-keys.rs:8: unexpected "error": '8:5: 8:19: expected a literal (or a UniCase if the unicase_support feature is enabled)'
error: tests/compile-fail-unicase/equivalent-keys.rs:9: unexpected "error": '9:5: 9:19: expected a literal (or a UniCase if the unicase_support feature is enabled)'
error: tests/compile-fail-unicase/equivalent-keys.rs:7: expected error not found: duplicate key UniCase("FOO")
error: tests/compile-fail-unicase/equivalent-keys.rs:8: expected note not found: one occurrence here
error: tests/compile-fail-unicase/equivalent-keys.rs:9: expected note not found: one occurrence here
```

**Comment by sfackler** [comment]
Ah, duh, right

File: phf_macros/tests/compiletest.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    let cfg_mode = mode.parse().ok().expect("Invalid mode");
+
+    config.mode = cfg_mode;
+    config.target_rustcflags = Some("-L target/debug/".to_owned());
+    config.src_base = PathBuf::from(format!("tests/{}", directory));
+
+    compiletest::run_tests(&config);
+}
+
+// #[test]
```
Comment (sfackler) [inline phf_macros/tests/compiletest.rs]: Dead code

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**